crates/core/src/transport.rs

```diff
@@ -81,27 +83,36 @@ struct StreamedMessagePart {
 }
 
 // todo: split this into type for handling inbound (encrypted)/outbound (decrypted) packets for clarity
-struct PacketData {
-    data: [u8; MAX_PACKET_SIZE],
+struct PacketData<const N: usize = MAX_PACKET_SIZE> {
+    data: [u8; N],
     size: usize,
 }
 
 // todo: this is temporal, we need to generate those
 static NONCE: [u8; 12] = [0; 12];
```

Comment:
This **must** be different for every packet but doesn't necessarily need to be random, eg. it can just be incremented by one each time. We could take advantage of this to avoid sending the entire nonce with each packet, but this mechanism needs to be robust against packet loss. Probably a premature optimization at this stage.